client/my-sites/stats/stats-post-anniversaries/index.jsx

```diff
@@ -0,0 +1,241 @@
+/*
+ * Post Anniversaries
+ *
+ * This component displays the post anniversaries based on the current day.
+ *
+ * There is no loading or empty state: the module is only displayed if there is
+ * at least one anniversary. That is the reason why this module should ideally
+ * be inserted at the end of a column.
+ */
+
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { localize, moment } from 'i18n-calypso';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import StatsModuleExpand from '../stats-module/expand';
+import SectionHeader from 'components/section-header';
+import PostsList from './posts-list';
+import QueryPosts from 'components/data/query-posts';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getSitePostsForQuery } from 'state/posts/selectors';
+import QueryPostStats from 'components/data/query-post-stats';
+import { getPostStats } from 'state/stats/posts/selectors';
+import { getSiteSlug } from 'state/sites/selectors';
+
+// Maximum number of simultaneous queries to fetch the posts by year
+const MAX_POST_QUERIES = 10;
+
+// Maximum number of posts to display
+const MAX_POSTS_DISPLAYED = 3;
+
+const PERIODS_ALLOWED = [ 'day', 'week', 'month' ];
+
+class StatModulePostAnniversaries extends Component {
+	render() {
+		const {
+			topPostsOnly,
+			oldestPostQuery,
+			postsByYearQueries,
+			postsByYear,
+			topPostsByYear,
+			siteId,
+			translate,
+			summary,
+			period,
+		} = this.props;
+
+		if ( ! siteId || PERIODS_ALLOWED.indexOf( period.period ) === -1 ) {
+			return null;
+		}
+
+		const allPosts = Array.prototype.concat( ...postsByYear );
+
+		let title = translate( 'Anniversaries', { comment: 'Title of the anniversaries module' } );
+		if ( topPostsOnly ) {
+			title = translate( 'Top %d anniversaries', {
+				args: MAX_POSTS_DISPLAYED,
+				comment: 'Title of the anniversaries module (when the maximum is reached)',
+			} );
+		}
+
+		return (
+			<div>
+				{ <QueryPosts siteId={ siteId } query={ oldestPostQuery } /> }
+				{ postsByYearQueries.map(
+					( query, i ) => <QueryPosts key={ i } siteId={ siteId } query={ query } />,
+				) }
+				{ ! allPosts.length
+					? null
+					: <div>
+							{ topPostsOnly
+								? allPosts.map(
```

Comment:
This worries me. Sure, HTTP2 requests are slimmer for the client, but won't the server still have to process these requests? How is this bounded?